### ISSUE TITLE
Your code does not seem to be runnable for the moment

### ISSUE THREAD

**Issue by xli458** [chenxn2020/GOSE#3]
(1) In your RE module in file "RE.py" line 349, the variable "logits" is not defined. 
(2) In the same function (i.e., forward()), the following functions are not defined:

self.get_multi_loss()
self.get_single_loss()

(3) In addition, function self.get_type_loss() seems to be incomplete (as shown below):

```
def get_type_loss(self,type_logits,key_mask,entities,relations):
        # logits 2,64,65,3
        logits = self.softmax(type_logits[:,:,0])
        B = logits.shape[0]
        device = logits.device
        key_mask = key_mask.bool()
        loss_fcn = CrossEntropyLoss()

        for b in range(B):
            logit = logits[b][key_mask[b]]

            from IPython import embed;embed()
        relations 
```

I have another question here for you as well:
In the RE module in file "RE.py", the maximum number of keys and values are predefined:

self.max_key = 64 (line 158)
self.max_value = 64 (line 159)

I anticipate that your implementation will use more VRAM compared to the original RE decoder in unilm implementation for LayoutXLM, and also, it will be slower as you even performed iterative learning. Have you benchmarked how much more VRAM is needed and how much slower your decoder is compared to the original RE decoder in unilm?
These are important considerations as it may mean that your approach cannot be effectively applied when the number of relations are many, which significantly limit its application to real-life problems. 

Thanks so much for the attention.

**Comment by xli458** [comment]
It seemed that that file is not actually called. Will check further. Ignore this issue for now
